src/main/keep-container/keep-content/keep-common-editor/keep-common-editor.component.ts

```diff
@@ -0,0 +1,136 @@
+import { Component, ElementRef, Inject, Input, OnInit, ViewChild } from '@angular/core';
+import { Note } from "../../../Data Types/Note";
+import { Observable, of, Subscription, switchMap } from "rxjs";
+import { NoteService } from "../../../Service/note.service";
+import { KeepCommonNoteComponent } from "../keep-common-note/keep-common-note.component";
+import { MAT_DIALOG_DATA, MatDialog, MatDialogRef } from "@angular/material/dialog";
+import { Label } from "../../../Data Types/Label";
+import { LabelService } from "../../../Service/label.service";
+
+@Component({
+  selector: 'app-keep-common-editor',
+  templateUrl: './keep-common-editor.component.html',
+  styleUrls: ['./keep-common-editor.component.scss'],
+
+})
+export class KeepCommonEditorComponent implements OnInit {
+
+  @Input() selectedNote!: Note | null; // Initialize as null
+  notes$: Observable<Note[]>;
+  labels: Label[] = [];
+  private labelListSubscription!: Subscription;
+  searchLabelText: string = '';
+  @ViewChild('formContainer') formContainer!: ElementRef;
+
+  handleMoreIconClick(event: MouseEvent, note: Note) {
+    event.stopPropagation();
+    note.showSelectedDropdown = !note.showSelectedDropdown;
+    note.isMoreIconClicked = !note.isMoreIconClicked;
+    note.showSelectedLabelDropdown = false;
+
+  }
+
+  hasLabels(note: Note) {
+    this.labelListSubscription = this.labelService.labelList$.subscribe((labels: Label[]) => {
+      this.labels = labels;
+    });
+    return note.labels.length > 0;
+  }
+
+  constructor(@Inject(MAT_DIALOG_DATA) public data: {
+    Note: Note,
+    dialogBoxOpen: boolean
+  }, public dialogRef: MatDialogRef<KeepCommonEditorComponent>, private noteService: NoteService, public dialog: MatDialog, private keepCommonNoteComponent: KeepCommonNoteComponent, private labelService: LabelService) {
+    this.notes$ = this.noteService.getNotes();
+  }
+
+  showLabelDropdown(event: Event, note: Note) {
+    event.stopPropagation();
+    note.showSelectedDropdown = !note.showSelectedDropdown;
+    note.showSelectedLabelDropdown = !note.showSelectedLabelDropdown;
+  }
+
+  ngOnInit(): void {
```

Comment:
I think we should follow the same practice for all the components:
1. @ViewChild/@Input/@Output
2. local variables
3. constructor 
4. ngOnInit/ngAfterViewInit etc
5. functions
6. ngOnDestroy

Reply:
updated all files with this pattern